coreneuron/io/reports/report_event.cpp

```diff
@@ -22,25 +23,44 @@ namespace coreneuron {
 ReportEvent::ReportEvent(double dt,
                          double tstart,
                          const VarsToReport& filtered_gids,
-                         const char* name)
+                         const char* name,
+                         double report_dt)
     : dt(dt)
     , tstart(tstart)
-    , report_path(name) {
+    , report_path(name)
+    , report_dt(report_dt) {
     VarsToReport::iterator it;
     nrn_assert(filtered_gids.size());
     step = tstart / dt;
+    reporting_period = static_cast<int>(report_dt / dt);
```

Comment:
but this is exactly what it is, the number of timesteps after the variable will be recorded. When this is multiple of the current step we do the summation (in any other case it doesn't make sense since it wont be recorded).